### DISCUSSION TITLE
Does ECJ support annotation processors out of the box without any configurations ?

### DISCUSSION THREAD

**Discussion by gayanper** [eclipse-jdt/eclipse.jdt.core#1683] [Q&A]
Given that annotation processors are available in the compile time classpath of a project, does ECJ capable of using those annotation processors to to output 

- Java classes
- AST Tree which is used for Completions and Diagnostics ?

**Comment by guw** [comment]
AFAIK The batch compiler supports `ServiceLoader` system for discovery (via `BatchAnnotationProcessorManager`). But the `BatchAnnotationProcessorManager` requires the processor path to be populated at the command line to support the lookup. It does not look into the classpath option.

Logic is here:
https://github.com/eclipse-jdt/eclipse.jdt.core/blob/e3e6fd35fb0e58009c80b47e6d3e21a2c93640c0/org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/apt/dispatch/BatchAnnotationProcessorManager.java#L130

**Comment by stephan-herrmann** [comment]
Please note the discussion in and around https://mail.openjdk.org/pipermail/jdk-dev/2024-May/009028.html
